receiver/mongodbatlasreceiver/logs.go

```diff
@@ -0,0 +1,309 @@
+// limitations under the License.
+
+package mongodbatlasreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbatlasreceiver"
+
+import (
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"errors"
+	"io"
+	"net"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"go.mongodb.org/atlas/mongodbatlas"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbatlasreceiver/internal"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbatlasreceiver/internal/model"
+)
+
+type logsReceiver struct {
+	log         *zap.Logger
+	cfg         *Config
+	client      *internal.MongoDBAtlasClient
+	consumer    consumer.Logs
+	stopperChan chan struct{}
+	wg          sync.WaitGroup
+}
+
+type resourceInfo struct {
+	Org      string
+	Project  mongodbatlas.Project
+	Cluster  mongodbatlas.Cluster
+	Hostname string
+	LogName  string
+	start    string
+	end      string
+}
+
+// MongoDB Atlas Documentation reccommends a polling interval of 5  minutes: https://www.mongodb.com/docs/atlas/reference/api/logs/#logs
+const collectionInterval = time.Minute * 5
+
+func newMongoDBAtlasLogsReceiver(settings component.ReceiverCreateSettings, cfg *Config) (*logsReceiver, error) {
+	client, err := internal.NewMongoDBAtlasClient(cfg.PublicKey, cfg.PrivateKey, cfg.RetrySettings, settings.Logger)
+	if err != nil {
+		return nil, err
+	}
+	recv := &logsReceiver{log: settings.Logger, cfg: cfg, client: client, stopperChan: make(chan struct{})}
+	return recv, nil
+}
+
+// Log receiver logic
+func (s *logsReceiver) Start(ctx context.Context, host component.Host) error {
+	s.wg.Add(1)
+	go func() {
+		defer s.wg.Done()
+		s.collect(ctx)
+	}()
+	return nil
+}
+
+func (s *logsReceiver) Shutdown(ctx context.Context) error {
+	close(s.stopperChan)
+	s.wg.Wait()
+	return s.shutdown(ctx)
+}
+
+func (s *logsReceiver) shutdown(context.Context) error {
+	return s.client.Shutdown()
+}
+
+// parseHostName parses out the hostname from the specified cluster host
+func parseHostName(s string, logger *zap.Logger) []string {
+	var hostnames []string
+
+	// first check to make sure string is not empty
+	if s == "" {
+		return []string{}
+	}
+
+	// create an array with a comma delimiter from the original string
+	tmp := strings.Split(s, ",")
+	for _, t := range tmp {
+
+		h := strings.TrimPrefix(t, "mongodb://")
+
+		// separate hostname from scheme and port
+		host, _, err := net.SplitHostPort(h)
+		if err != nil {
+			logger.Error("Could not parse out hostname: " + host)
+			continue
+		}
+		// hostname parsed successfully
+		hostnames = append(hostnames, host)
+	}
+
+	return hostnames
+}
+
+// KickoffReceiver spins off functionality of the receiver from the Start function
+func (s *logsReceiver) collect(ctx context.Context) {
+	resource := resourceInfo{start: strconv.Itoa(int(time.Now().Unix())), end: ""}
```

Comment:
```suggestion
	resource := resourceInfo{start: strconv.Itoa(int(time.Now().Unix())), end: ""}
```

I get that this allows us to format the timestamp only once, but typically times should be kept as a native representation of time (eg `time.Time`) unless there is a meaningful performance impact. Since we're collecting every 5 minutes by default, it is negligible to do this as needed.

The actual conversion should be pulled into a function too.